### PR TITLE
feat(transfer): use protobuf for packet data v2 serialisation

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

closes: #6630 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/apps/transfer/keeper/relay_test.go [hunk 1]
```diff
@@ -1154,55 +1154,9 @@ func (suite *KeeperTestSuite) TestPacketForwardsCompatibility() {
 		expError    error
 		expAckError error
 	}{
-		{
```
[the remark this answers is not in the document]
Comment (crodriguezvega) [reply modules/apps/transfer/keeper/relay_test.go:1157]: Sorry, can you elaborate why these are not needed now?

**Review by crodriguezvega — APPROVED** [review]
Nice!

File: modules/apps/transfer/keeper/relay_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (srdtrk) [reply modules/apps/transfer/keeper/relay_test.go:1157]: This was discussed and decided that we should disable. This is the whole reason we did that funny custom json unmarshaling.

File: modules/apps/transfer/internal/packet_test.go [hunk 2]
```diff
@@ -73,6 +79,57 @@ func TestUnmarshalPacketData(t *testing.T) {
...
+		{
+			"success",
+			func() {},
+			nil,
+		},
+		{
+			"failure: new field present in packet data",
+			func() {
+				// packet data containing extra field unknown to current proto file.
+				packetDataBz = []byte("\n%\n\x1d\n\x04atom\x1a\x15\n\btransfer\x12\tchannel-0\x12\x041000\x12\x06sender\x1a\breceiver*\x002\tnew_value")
```
Comment (colin-axner) [inline modules/apps/transfer/internal/packet_test.go:101]: could be nice to take `packetDataBz` and append the new field for readability

**Pull request merged**